docs/modules/ROOT/pages/references/architecture/control-api-design.adoc

```diff
@@ -0,0 +1,181 @@
+These resources are similar to views in a relational database.
+The benefit of providing these resources instead of only using CRDs is that we can calculate access permissions dynamically for every request.
+The same concept is also used by OpenShift with its `Project` resource which represents RBAC filter `Namespaces` (see https://github.com/openshift/kube-projects[kube-projects]).
+
+=== Authentication and Authorization
+
+Authentication against the API server is done by the {idp} by leveraging OIDC.
+This allows to transport authorized information about the user in a JWT token.
+
+For authorization, standard https://kubernetes.io/docs/reference/access-authn-authz/rbac/[Kubernetes RBAC] is being used.
+Kyverno policies can be used to implement enhanced policies, for example the number of resources of a specified kind a user is allowed to create.
+
+== Main Use-Cases
+
+All use-cases are from the view of an API user.
+
+Authentication and Authorization::
+* Authenticate against the API, so that the API knows who I am
+* Authorization to work with resources granted to me
+
+Organizations::
+* List organizations I have access to
+* Create a new organization if I'm allowed to
+* Update an existing organization if I'm allowed to
+* Delete an existing organization if I'm allowed to
+* Invite a user to an organization if I'm allowed to send invitations
+* Join an existing organization If I'm invited
+
+Zones::
+* List of zones I have access to
+* See details of zone I have access to
+
+== Global Resources
+
+These are the resources which are global to the {product} instance.
+They are available in the global scope of the Kubernetes API server.
+
+=== `Organization` (Virtual)
+
+Organizations are the container for all resources which belong to an organization.
+It is represented by a Kubernetes Namespace with specific labels and annotations.
+This allows to use standard Kubernetes RBAC rules for organization-specific resources.
+
+To circumvent the current limitation of Kubernetes to show filtered lists of cluster-scoped resources based on access rights, the virtual object `Organization` represents an organization.
+
+The dynamic filter works through the list of namespaces which represent an organization and checks if the subject has access to the namespace.
+
+[source,yaml]
+----
+apiVersion: appuio.io/v1
+kind: Organization
+metadata:
+  name: acme-corp
+spec:
+  displayName: Acme Corp.
+----
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: org-acme-corp <1>
+  labels:
+    appuio.io/resource-type: organization <2>
+    organization.appuio.io/displayName: Acme Corp. <3>
+----
+<1> Prefixed (`org`) name to circumvent possible name collision (e.g. if a company is called "kube-system")
+<2> Identify resource type, used by the API server to filter for namespace representing organizations
+<3> Reflected in the organization object as `spec.displayName`
```

Comment:
This part is missing a bit of context? (please rephrase if I understood it wrong)
```suggestion
The dynamic filter works through the list of namespaces which represent an organization and checks if the subject has access to the namespace.

Consider a user-generated `Organization`:
[source,yaml]
----
apiVersion: appuio.io/v1
kind: Organization
metadata:
  name: acme-corp
spec:
  displayName: Acme Corp.
----

The API server would then create a `Namespace` with following spec:
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  name: org-acme-corp <1>
  labels:
    appuio.io/resource-type: organization <2>
    organization.appuio.io/displayName: Acme Corp. <3>
----
<1> Prefixed (`org`) name to circumvent possible name collision (e.g. if a company is called "kube-system")
<2> Identify resource type, used by the API server to filter for namespace representing organizations
<3> Reflected in the organization object as `spec.displayName`
```